File: LostCities/Enums.cs

namespace LostCities;

/// <summary>
/// Identifies a player, or <see cref="None"/>.
/// </summary>
public enum Player
{
    None = 0,
    Player1 = 1,
    Player2 = 2
}

/// <summary>
/// Identifies an expedition color.
/// </summary>
public enum Color
{
    Red = 1,
    Green,
    Blue,
    White,
    Yellow,
    Purple
}

/// <summary>
/// Either a numerical value for a card, or <see cref="Wager"/>.
/// </summary>
public enum Value
{
    Wager = 0,
    Two = 2,
    Three,
    Four,
    Five,
    Six,
    Seven,
    Eight,
    Nine,
    Ten
}

File: LostCities.Viewer/src/enums.ts
export enum CardColor {
    RED = 1,
    GREEN,
    BLUE,
    WHITE,
    YELLOW,
    PURPLE
}

export enum CardValue {
    WAGER = 0,
    TWO = 2,
    THREE,
    FOUR,
    FIVE,
    SIX,
    SEVEN,
    EIGHT,
    NINE,
    TEN
}

export enum Player {
    NONE,
    PLAYER1,
    PLAYER2
}

export const CARD_SYMBOLS = new Map([
    [CardColor.RED, "♥"],
    [CardColor.GREEN, "♠"],
    [CardColor.BLUE, "♣"],
    [CardColor.WHITE, "♦"],
    [CardColor.YELLOW, "★"],
    [CardColor.PURPLE, "⚘"]
]);